lib/cc/cli/source_fingerprint.rb

```diff
@@ -0,0 +1,69 @@
+require "digest/md5"
+
+module CC
+  module CLI
+    class SourceFingerprint
+      def initialize(output)
+        @output = output
+      end
+
+      def compute
+        md5 = Digest::MD5.new
+        md5 << key
```

Comment:
That's true. The new behavior will produce more issues, and potentially cause more failing status checks, but I think that's a favorable outcome. We want to reveal issues, not sweep them under the rug.

If an engine author wishes to override the default fingerprinting behavior, they can easily do so by providing that key in the issue, but I think it's a more useful default that what we have now.